examples/asyncio_displayio_button.py

```diff
@@ -0,0 +1,312 @@
+    y=BUTTON_3_Y,
+    width=BUTTON_WIDTH,
+    height=BUTTON_HEIGHT,
+    style=BUTTON_STYLE,
+    fill_color=BUTTON_FILL_COLOR,
+    outline_color=BUTTON_OUTLINE_COLOR,
+    label=BUTTON_3_LABEL,
+    label_font=terminalio.FONT,
+    label_color=BUTTON_LABEL_COLOR,
+)
+
+# Button state data object. Will hold either true of false whether button is currently pressed
+class ButtonState:
+    # pylint: disable=too-few-public-methods
+    def __init__(self, initial_state):
+        self.state = initial_state
+
+
+# Interval length data object. Holds the amount of time in ms the interval should last for
+class Interval:
+    # pylint: disable=too-few-public-methods
+    def __init__(self, initial_value):
+        self.value = initial_value
+
+
+# main group to show things on the display
+main_group = displayio.Group()
+
+# Initialze first circle
+palette_1 = displayio.Palette(2)
+palette_1[0] = 0x125690
+palette_1[1] = 0x125690
+circle_1 = vectorio.Circle(pixel_shader=palette_1, radius=15, x=20, y=20)
+
+# Initialze second circle
+palette_2 = displayio.Palette(2)
+palette_2[0] = 0x12FF30
+palette_2[1] = 0x12FF30
+circle_2 = vectorio.Circle(pixel_shader=palette_2, radius=15, x=60, y=20)
+
+# add everything to the group, so it gets displayed
+main_group.append(circle_1)
+main_group.append(circle_2)
+main_group.append(invert_color_btn)
+main_group.append(interval_slower_btn)
+main_group.append(interval_faster_btn)
+
+
+async def blink(palette, interval, count, button_state):  # Don't forget the async!
+    """
+    blink coroutine. Hides and shows a vectorio shape by
+    using make_transparent() and make_opaque() on it's palette.
+
+    :param displayio.Palette palette: The palette to change colors on for blinking
+    :param Interval interval: The Interval data object containing the interval length to use
+    :param int count: The number of times to repeat the blink. -1 for indefinite loop
+    :param ButtonState button_state: The ButtonState data object for the invert color button
+    """
+    while count < 0 or count > 0:
+
+        # if the color button is pressed
+        if button_state.state:
+            # if the color is still on default
+            if palette[0] == palette[1]:
+                # invert the color by subtracting from white
+                palette[0] = 0xFFFFFF - palette[0]
+
+        # if the color button is not pressed
+        else:
+            # set the color back to default
+            palette[0] = palette[1]
+
+        # hide the circle
+        palette.make_opaque(0)
+        # wait interval length
+        await asyncio.sleep(interval.value / 1000)  # Don't forget the await!
+
+        # show the circle
+        palette.make_transparent(0)
+        # wait interval length
+        await asyncio.sleep(interval.value / 1000)  # Don't forget the await!
+
+        # decrement count if it's positive
+        if count > 0:
+            count -= 1
+
+
+async def monitor_color_button(button, button_state):
```

Comment:
Is there any way to prevent this coroutine from reading touches on the other buttons? I think it would be better to have a single coroutine reading the touch screen events.

Reply:
The latest commit refactors the touch handling into a single coroutine that calls the separate handle functions for each of the button types.

Reply:
LGTM!